packages/ui/components/form-core/src/choice-group/CustomChoiceGroupMixin.js

```diff
@@ -0,0 +1,204 @@
+import { dedupeMixin } from '@open-wc/dedupe-mixin';
+import { ChoiceGroupMixin } from './ChoiceGroupMixin.js';
+
+/**
+ * @typedef {import('../../types/choice-group/CustomChoiceGroupMixinTypes.js').CustomChoiceGroupMixin} CustomChoiceGroupMixin
+ * @typedef {import('../../types/choice-group/CustomChoiceGroupMixinTypes.js').CustomChoiceGroupHost} CustomChoiceGroupHost
+ */
+
+/**
+ * @param {any|any[]} value
+ * @returns {any[]}
+ */
+function normalizeArray(value) {
+  return Array.isArray(value) ? value : [value];
+}
+
+/**
+ * Extends the ChoiceGroupMixin to add optional support for custom user choices without altering the initial choice list.
+ *
+ * @type {CustomChoiceGroupMixin}
+ * @param {import('@open-wc/dedupe-mixin').Constructor<import('lit').LitElement>} superclass
+ */
+const CustomChoiceGroupMixinImplementation = superclass =>
+  // @ts-ignore https://github.com/microsoft/TypeScript/issues/36821#issuecomment-588375051
+  class CustomChoiceGroupMixin extends ChoiceGroupMixin(superclass) {
+    static get properties() {
+      return {
+        allowCustomChoice: {
```

Comment:
Considering the fact that this is designed to cover for multiple scenarios (including radiogroups), allowCustomChoice might be preferable over requireOptionMatch.

Let me give it a second thought, I will come back to it later :)

Reply:
Can you maybe give us an idea how this mixin would be used outside the combobox? With LionRadioGroup for instance?

Reply:
Your previous explanation makes sense, but indeed that is what I was thinking about the `allowCustomChoice`, and honestly I was expecting someone to comment on the way I went around it, inverting the value via property, as I didn't know about the intention for `requireOptionMatch = true`, and I didn't want to rename it completely as not to break compatibility (though I considered deprecating it, but that's not me to decide it).

I don't have a strong opinion either way other than `allowCustomChoice` matches better with the mixin's name. Let me know once you have a decision and I will gladly update it.

Cheers.

Reply:
Great, makes sense. I must say that I like `allowCustomChoice` better (it's easier to read), so happy to go with that (especially now it makes more sense in a broader perspective) :)

It would maybe be nice to make a small example (can be in the test of the CustomChoiceGroupMixin or as a demo of the mixin) with a radiogroup/checkbox/listbox with a custom text field.

If it means CustomChoiceGroupMixin can be applied to all ChoiceGroups, we might as well consider merging the allowCustomChoice functionality into ChoiceGroupMixin?

Reply:
> It would maybe be nice to make a small example (can be in the test of the CustomChoiceGroupMixin or as a demo of the mixin) with a radiogroup/checkbox/listbox with a custom text field.
> If it means CustomChoiceGroupMixin can be applied to all ChoiceGroups, we might as well consider merging the allowCustomChoice functionality into ChoiceGroupMixin?

The way it's implemented currently we need to add the mixin to those other controls so I can make those examples. Makes sense to merge that to ChoiceGroupMixin, that's just raise the question: Do we want all the users have to handle Custom Choices every time a new ChoiceGroupMixin is created?

Also, do want that done in this one PR?
Also can I assume that renaming the property is not included?  Or this will break current users?